addOns/exim/src/test/java/org/zaproxy/addon/exim/har/HarImporterUnitTest.java

```diff
@@ -126,11 +164,137 @@ void shouldCompleteListenerIfFileNotFound(@TempDir Path dir) throws Exception {
+        // When
+        HarLog harLog =
+                new HarReader()
+                        .readFromString(
+                                this.getHar(
+                                        getResourcePath("templates/missingRespHttpVer.har"),
+                                        Map.of("replace", "")))
+                        .getLog();
+        HarImporter importer = new HarImporter(harLog, listener);
+        // Then
+        assertThat(logMessages.size(), is(1));
+        assertThat(importer.isSuccess(), equalTo(true));
+        verify(listener).completed();
+        System.out.println(logMessages.get(0));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"h3", "http/3"})
+    void shouldDropHarEntryIfHttpVersionNotSupported(String version) throws Exception {
+        // Given
+        ProgressPaneListener listener = mock(ProgressPaneListener.class);
+        // When
+        HarLog harLog =
+                new HarReader()
+                        .readFromString(
+                                this.getHar(
+                                        getResourcePath("templates/missingReqHttpVer.har"),
+                                        Map.of("replace", version)))
+                        .getLog();
+        HarImporter importer = new HarImporter(harLog, listener);
+        // Then
+        assertThat(logMessages.size(), is(1));
+        assertThat(importer.isSuccess(), equalTo(true));
+        verify(listener).completed();
+    }
+
     private static HarLog createHarLog(HttpMessage message) {
         HarLog harLog = HarUtils.createZapHarLog();
-        HarEntries harEntries = new HarEntries();
-        harEntries.addEntry(HarUtils.createHarEntry(message));
+        List<HarEntry> harEntries = new ArrayList<>();
+        harEntries.add(HarUtils.createHarEntry(message));
         harLog.setEntries(harEntries);
         return harLog;
     }
+
+    /**
+     * Gets the contents of the file with the given path, replaced with the given parameters.
+     *
+     * @param resourcePath the path to the resource.
+     * @param params the parameters to replace in the contents, might be {@code null}.
+     * @return the contents of the file.
+     * @see TestUtils#getResourcePath(String)
+     */
+    public String getHar(Path file, Map<String, String> params) {
+        try {
+            String har = new String(Files.readAllBytes(file), StandardCharsets.UTF_8);
+            if (params != null) {
+                // Replace all of the supplied parameters
+                for (Entry<String, String> entry : params.entrySet()) {
+                    har = har.replaceAll("@@@" + entry.getKey() + "@@@", entry.getValue());
+                }
+            }
+            return har;
+        } catch (IOException e) {
+            System.err.println("Failed to read file " + file.toAbsolutePath());
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
This is a slight modification of the TestUtils getHtml method because when I used it, it would claim the path was null when it definitely wasn't 😞

Reply:
The `getHtml` seems to be working properly, the only `null` I'm seeing is in the `persistMessage` method (which is expected as it needs some more mocking).